### PR TITLE
Enable xqueue-watcher to handle python3 queues.

### PR DESCRIPTION
**Pull request by jcdyer**
* Make grader_support package compatible with python 2 and 3
* Add tests for python3
* Update travis to install a python3 virtualenv for tests.

## Reviewers

- [x] @efischer19 
- [ ] @maxrothman

### CODE REVIEW

**Comment by jcdyer** [comment]
Tests pass:
```
(xqw)dhcp-18-111-109-76:xqw cliffdyer$ nosetests tests
...........................
```

Caveat is that it is looking for python3 at /usr/bin/ or /usr/local/bin/python3, and that python3 needs to have six installed.  It would be better to give it access to a python3 virtualenv, but I'm not sure of a reasonable way to do it that will be robust for testing on different computers, in different directories, etc.  Maybe we need an environment variable for XQUEUEWATCHER_PYTHON3_VENV or something.

**Comment by jcdyer** [comment]
I created the environment variable, and configured travis to create a py3 virtualenv to use for testing. Had to update the travis build box though, because my python 2/3 compatible code was apparently not compatible with 3.2.

File: grader_support/gradelib.py [hunk 1]
```diff
@@ -71,7 +75,7 @@ def __init__(self):
         # in the specified order.  (foldl)
         self._preprocessors = [fix_line_endings]
 
-        # how many EndTest's did we raise and not catch? So we can make sure the student 
+        # how many EndTest's did we raise and not catch? So we can make sure the student
```
Comment (efischer19) [inline grader_support/gradelib.py]: Nit: the apostrophe in `EndTest's` is incorrect

**Comment by efischer19** [comment]
One tiny nit, 👍 otherwise. Good test coverage!

**Comment by jcdyer** [comment]
And it wasn't even a line I wrote!

File: .travis.yml [hunk 2]
```diff
@@ -1,11 +1,15 @@
+dist: trusty
```
Comment (maxrothman) [inline .travis.yml]: You should also test this on xenial since that's what we're running in prod

**Comment by maxrothman** [comment]
👍

File: .travis.yml (hunk 2 above)
Comment (jcdyer) [inline .travis.yml]: Note from our conversation: Travis doesn't yet support builds on Xenial, and it doesn't seem to be a high priority for them right now.

**Pull request merged**